app/controllers/comments_controller.rb

```diff
@@ -1,22 +1,40 @@
 # frozen_string_literal: true
 
 class CommentsController < ApplicationController
-  before_filter :set_comment, only: [:show, :update, :destroy]
-  before_filter :set_article, only: [:index, :create]
+  before_action :set_comment, only: [:show, :update, :destroy]
+  before_action :set_article, only: [:index, :create]
 
   def index
+    @comments = @article.comments
+    render json: { comments: @comments }
   end
 
   def show
+    render json: @comment
   end
 
   def create
+    @comment = Comment.new(comment_params)
+    @comment.article = @article
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: @comment.errors, status: :unprocessable_entity
+    end
   end
 
   def update
+    if @comment.update(comment_params)
+      render json: @comment, status: :ok
```

Comment:
The older tests wanted a json response. We need to update the tests and then the code to just expect status: :ok